### DISCUSSION TITLE
End-of-life date for Remix v2?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [remix-run/remix#10343] [Q&A] [answered]
Will Remix v2 continue to be supported for a bit? Will it get updates for React 19 and/or Vite 6, or will we be forced to migrate to RR7 first?

Honestly, RR7 doesn't feel ready for migration yet. The documentation is very sparse compared to Remix; the effort required appears to be a lot higher than advertised; and for now there's little payoff once we're done, especially since we use APIs like `useRouteLoaderData` that don't appear to have modern type-generated replacements available yet. I have no doubt it will be a worthwhile upgrade in the end, but it's hardly a "boring" migration, at least not yet.

At the same time, React 19 brings immediate improvements in developer experience, like diffs for hydration errors. It also provides new APIs for rendering meta and stylesheet tags that could help us simplify big chunks of some of our routes, which could in turn help simplify our migration to RR7 by reducing our need to pass data between routes in the first place.

So, any chance we'll get some continued support for Remix while RR7 begins to mature?

**Comment by DavidGutierrez94** [comment]
Hi! is there any update on this?

**Comment by brookslybrand** [comment, marked as answer]
@aaronadamsCA Remix v2 will receive critical security updates and features/bug fixes that help with with upgrading to React Router v7. I don't have an exact EOL date for you right now.

Agreed that the React Router docs need a bit more work. We are working on adding all of main APIs back into the main docs. We were trying to make the [autogenerated API reference](https://api.reactrouter.com/v7/) the primarily source of truth for much of our docs, but unfortunately it is not measuring up to our standards

However, from a code standpoint, we believe v7 is completely ready for Remix users, and many have already upgraded with ease (assuming all future flags were enabled). To take your specific example, `useRouteLoaderData` still exists and works exactly like it did before, even though we have plans to add improved APIs that are better integrated with our typegen.

If there's anything else making it a "not boring" upgrade, we'd love the feedback.
